### ISSUE TITLE
[TSQL] Unparsable section when create table with `SYSTEM_VERSIONING`, parameter `HISTORY_RETENTION_PERIOD`

### ISSUE THREAD

**Issue by r-petit** [sqlfluff/sqlfluff#5271]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Having an SQL script that create a table with the `SYSTEM_VERSIONING` and parameters `HISTORY_RETENTION_PERIOD` throws an unparsable section error.

### Expected Behaviour

We should not have an error of parsing when using this parameter.

### Observed Behaviour

According to the Microsoft documentation for the [CREATE TABLE](https://bit.ly/3RQazXQ), only `HISTORY_TABLE` and `DATA_CONSISTENCY_CHECK` are allowed as parameters. But also according to the documentation and examples provided in how to [manage historical data in temporal tables](https://bit.ly/3LMOyFA), we can provide the parameter `HISTORY_RETENTION_PERIOD`.

To fix this we just need to add it as optional in this file [dialect_tsql.py](https://github.com/sqlfluff/sqlfluff/blob/5680394b82888efc6d63656a48e049caf27029f8/src/sqlfluff/dialects/dialect_tsql.py#L1388C32-L1388C32).

### How to reproduce

```sql
CREATE TABLE [dbo].[PARSING_ERROR] (
	[id] INT NOT NULL
) ON [PRIMARY] WITH (
	SYSTEM_VERSIONING = ON (
		HISTORY_TABLE = [cpt].[CATEGORIES_HISTORY],
		HISTORY_RETENTION_PERIOD = INFINITE,
		DATA_CONSISTENCY_CHECK = ON
	)
);
```

### Dialect

tsql

### Version

```zsh
$ sqlfluff --version
sqlfluff, version 2.3.2

$ python --version
Python 3.11.5
```

### Configuration

Default configuration when running the following command:

```zsh
$ sqlfluff lint --dialect tsql file.sql
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
